zenoh-jni/src/scouting.rs

```diff
@@ -21,6 +21,7 @@ use jni::{
 };
 use zenoh::{config::WhatAmIMatcher, prelude::Wait};
 use zenoh::{scouting::Scout, Config};
+use zenoh_protocol::core::ZenohIdProto;
```

Comment:
Why `ZenohIdProto` and not `ZenohId`?

Reply:
In general all operations at API-level shouldn't require importing protocol-level types.